### PR TITLE
Custom image upload restrictions

### PR DESCRIPTION
**Pull request by objecttothis**
This PR makes image restrictions customizable by the administrator in Settings->General. It also fixes a bug causing the image remove to not work.

### CODE REVIEW

**Comment by objecttothis** [comment]
Turned out to be easy to fix the bug with images on items not deleting.
The original code had " with ' inside:
`			url: "<?php echo site_url('$controller_name/remove_logo/$item_info->item_id'); ?>",`
but needed to be reversed because PHP doesn't parse variable names inside of ' but will inside of "
`			url: '<?php echo site_url("$controller_name/remove_logo/$item_info->item_id"); ?>',`

**Comment by objecttothis** [comment]
@jekkos the demo seems to be broken again.  Do you think it's the same issue?

**Comment by jekkos** [comment]
Demo seems fine, it restarts every hour so maybe you checked it during a restart?

File: application/controllers/Config.php [hunk 1]
```diff
@@ -210,15 +210,23 @@ public function index()
...
+
 		// load all the license statements, they are already XSS cleaned in the private function
 		$data['licenses'] = $this->_licenses();
 		// load all the themes, already XSS cleaned in the private function
 		$data['themes'] = $this->_themes();
-		
-		$data['mailchimp'] = array();
+
+		//Load General related fields
+		$image_allowed_types 			= array('jpg','jpeg','gif','svg','webp','bmp','png','tif','tiff');
```
Comment (jekkos) [inline application/controllers/Config.php:222]: minor indentation issue here

**Comment by andriux1990** [comment]
@objecttothis A question. Now, if I wanted to change the images to a higher resolution ... where will I have to make the change?.

since this option has been removed from the item.

**Comment by objecttothis** [comment]
@andriux1990 I wasn't aware that the change image button had been removed. If you edit an item that has an image the change image button doesn't appear next to remove image?

**Comment by andriux1990** [comment]
sorry .. everything is fine .. and thank you very much for the changes .. i like it more like this
Screenshot_20200513_090028

**Comment by objecttothis** [comment]
@andriux1990 we need to get that AdminLTE interface into the master!  That is sexy.

That aside, I wasn't sure if the Bootstrap Icons/Tooltips would be descriptive enough, but it is way more streamlined to do it that way than in other areas.

**Comment by andriux1990** [comment]
> @andriux1990 we need to get that AdminLTE interface into the master! That is sexy.
> 
> That aside, I wasn't sure if the Bootstrap Icons/Tooltips would be descriptive enough, but it is way more streamlined to do it that way than in other areas.

ospos does not get along well with adminLTE

**Comment by andriux1990** [comment]
> @andriux1990 we need to get that AdminLTE interface into the master! That is sexy.
> 
> That aside, I wasn't sure if the Bootstrap Icons/Tooltips would be descriptive enough, but it is way more streamlined to do it that way than in other areas.

Here I have played a bit .. you can see how it looks
http://45.181.68.3:8080/~andu/osPosLTE/public

**Comment by objecttothis** [comment]
> 
> 
> > @andriux1990 we need to get that AdminLTE interface into the master! That is sexy.
> > That aside, I wasn't sure if the Bootstrap Icons/Tooltips would be descriptive enough, but it is way more streamlined to do it that way than in other areas.
> 
> ospos does not get along well with adminLTE
> I'm trying to get them to get along

@andriux1990 I logged into the demo you posted earlier and it looks like there are a few minor quirks, but with a little massaging you or someone should be able to get it to work.  I assume you are using the free AdminLTE Bootstrap template and not one of the paid ones?

**Comment by objecttothis** [comment]
> 
> 
> > > @andriux1990 we need to get that AdminLTE interface into the master! That is sexy.
> > > That aside, I wasn't sure if the Bootstrap Icons/Tooltips would be descriptive enough, but it is way more streamlined to do it that way than in other areas.
> > 
> > 
> > ospos does not get along well with adminLTE
> > I'm trying to get them to get along
> 
> @andriux1990 I logged into the demo you posted earlier and it looks like there are a few minor quirks, but with a little massaging you or someone should be able to get it to work. I assume you are using the free AdminLTE Bootstrap template and not one of the paid ones?
> 
> I looked at it in Firefox Developer edition on my desktop and Firefox Beta on my Android 10 device and both looked pretty decent.

Are the issues with they way they look caused by the code in our views or by AdminLTE?  Ideally, we wouldn't have to customize the AdminLTE code, but instead change the views and the way they are written.  I say this because that will make future upgrades of the AdminLTE module easier.

**Comment by andriux1990** [comment]
so if you looked at it ... remove the link because there were some things that were not there yet

**Comment by andriux1990** [comment]
> > > > @andriux1990 we need to get that AdminLTE interface into the master! That is sexy.
> > > > That aside, I wasn't sure if the Bootstrap Icons/Tooltips would be descriptive enough, but it is way more streamlined to do it that way than in other areas.
> > > 
> > > 
> > > ospos does not get along well with adminLTE
> > > I'm trying to get them to get along
> > 
> > 
> > @andriux1990 I logged into the demo you posted earlier and it looks like there are a few minor quirks, but with a little massaging you or someone should be able to get it to work. I assume you are using the free AdminLTE Bootstrap template and not one of the paid ones?
> > I looked at it in Firefox Developer edition on my desktop and Firefox Beta on my Android 10 device and both looked pretty decent.
> 
> Are the issues with they way they look caused by the code in our views or by AdminLTE? Ideally, we wouldn't have to customize the AdminLTE code, but instead change the views and the way they are written. I say this because that will make future upgrades of the AdminLTE module easier.



> @andriux1990 I logged into the demo you posted earlier and it looks like there are a few minor quirks, but with a little massaging you or someone should be able to get it to work. I assume you are using the free AdminLTE Bootstrap template and not one of the paid ones?
> 
> I looked at it in Firefox Developer edition on my desktop and Firefox Beta on my Android 10 device and both looked pretty decent.

Like what peculiarities???
in this demo the kit works as a single product.

And this done in this adminlte3 https://adminlte.io/themes/v3/index.html

**Comment by objecttothis** [comment]
> Like what peculiarities???

The only one I remember now is in the reports, the footer appeared high up on the screen rather than down at the bottom.  There were a few others, but they all looked minor.  This is why it sounds to me like this could be made to work, but you said that it doesn't play well with OSPOS so I wasn't sure if I missed something.

**Pull request merged**